File: internal/indexer/testdata/ts/models/criticalUserModel.ts
/** BaseModel provides base entity identity fields. */
export class BaseModel {
  id: number;
  createdAt: string;

  constructor(id: number = 0) {
    this.id = id;
    this.createdAt = new Date().toISOString();
  }
}

/** CriticalUserModel is a heavily-used user representation that many files depend on. */
export class CriticalUserModel extends BaseModel {
  name: string;
  email: string;

  constructor(name: string, email: string) {
    super(0);
    this.name = name;
    this.email = email;
  }

  validate(): boolean {
    return this.name.length > 0 && this.email.includes("@");
  }

  display(): string {
    return `${this.name} <${this.email}>`;
  }
}

/** CriticalUserModelOptions describes creation options. */
export type CriticalUserModelOptions = {
  name: string;
  email: string;
};

/** createCriticalUserModel is a factory function. */
export function createCriticalUserModel(opts: CriticalUserModelOptions): CriticalUserModel {
  return new CriticalUserModel(opts.name, opts.email);
}

File: internal/indexer/testdata/python/models/critical_user_model.py
"""Critical user model module."""


class BaseModel:
    """BaseModel provides base entity identity fields."""

    def __init__(self, id: int = 0):
        self.id = id
        self.created_at = ""


class CriticalUserModel(BaseModel):
    """CriticalUserModel is a heavily-used user representation that many files depend on."""

    def __init__(self, name: str, email: str):
        super().__init__(0)
        self.name = name
        self.email = email

    def validate(self) -> bool:
        return bool(self.name) and "@" in self.email

    def display(self) -> str:
        return f"{self.name} <{self.email}>"


def create_critical_user_model(name: str, email: str) -> CriticalUserModel:
    """Factory function for creating CriticalUserModel instances."""
    return CriticalUserModel(name, email)
